### ISSUE TITLE
Progress bar with autoincrement set to true

### ISSUE THREAD

**Issue by mbahoshy** [minhtranite/react-progress-bar-plus#4]
When you initialize component like so "ProgressBar autoIncrement={true} " progress bar will not animate. However, if you initialize to false, and then change autoIncrement to true (through setState or other method) autoincrement works.

I think the issue here is that you are checking for autoincrement prop on componentWillReceiveProps. Prop should also be checked on componentDidMount

**Comment by minhtranite** [comment]
> When you initialize component like so "ProgressBar autoIncrement={true} " progress bar will not animate

Yes. process bar only show when percent >= 0 or < 100. Default percent = -1.

> if you initialize to false, and then change autoIncrement to true (through setState or other method) autoincrement works.

No. if percent still = `-1` progress bar still don't show.

You can see example: http://codepen.io/vn38minhtran/pen/adZVYQ

**Comment by mbahoshy** [comment]
This is not working for me ... In the example below I am trying to do a very simple implementation, setting percent={10} and autoIncrement={true}

http://codepen.io/anon/pen/QyGqKZ   - initializes autoIncrement={true}, does not work

But if i initially set autoIncrement={false}, then trigger re-render through setState({autoIncrement: true}) it starts working

http://codepen.io/anon/pen/BjQwQr  - initializes autoIncrement={false}, then click button to setState, does work

**Comment by minhtranite** [comment]
What is OS and browser you use ?
I run your `pen` on my browser it work.

**Comment by mbahoshy** [comment]
I tried in chrome, firefox, and safari ... running on a mac (whatever the latest os is). For the first codepen that I posted (http://codepen.io/anon/pen/QyGqKZ), the progress bar shows up, but does not autoIncrement.

To fix this, I took the js from your componentWillReceiveProps, and made it into a function named checkProps. Then i called this function on both componentWillReceiveProps and componentDidMount. Here is a gist of what is working for me (had to move a few things around to get it to build in my webpack bundle)
https://gist.github.com/mbahoshy/59871864627de48fd7c1
